src/Client/Pages/Identity/Users/Users.razor.cs

```diff
@@ -64,7 +64,7 @@ protected override async Task OnInitializedAsync()
                     || user.Email?.Contains(searchString, StringComparison.OrdinalIgnoreCase) == true
                     || user.PhoneNumber?.Contains(searchString, StringComparison.OrdinalIgnoreCase) == true
                     || user.UserName?.Contains(searchString, StringComparison.OrdinalIgnoreCase) == true,
-            createFunc: user => UsersClient.CreateAsync(user),
+            createFunc: user => UsersClient.CreateAsync("root",user),
```

Comment:
This is not right... should be the current tenant (you can get that from the user's claims).

Ideally we should have a separate method on the api side for creating a new user from the admin section here. That one would not be anonymous, and would also have a permission attached. It would then also not need the tenantId.

Reply:
dam - left that "root" in accidently when i couldn't figure out where to get it from - so you say I can get it from the user.Claims collections?

Reply:
Yes, but that's only when working with jwt auth. With AzureAd auth that's not available as a claim there... gonna have to create a new api endpoint... but that's the cleanest way anyway with the permissions and all... 

I would suggest renaming the current CreateUser to RegisterUser, and then add a new CreateUser that doesn't allow anonymous and also doesn't require the tenant header. They do basically the same thing, but are decorated differently:

```c#
    [HttpPost]
    [AllowAnonymous]
    [TenantIdHeader]
    [OpenApiOperation("Register a new user.", "")]
    public Task<string> RegisterAsync(CreateUserRequest request)
    {
        return _userService.CreateAsync(request, GetOriginFromRequest());
    }

    [HttpPost]
    [MustHavePermission(FSHAction.Create, FSHResource.Users)]
    [OpenApiOperation("Create a new user.", "")]
    public Task<string> CreateAsync(CreateUserRequest request)
    {
        return _userService.CreateAsync(request, GetOriginFromRequest());
    }
```

Reply:
OTOH, I just realize we don't need the create user functionality when working with azureAd (the users get automatically created on first login)... so I guess for now you could just take it from principal.GetTenantId()...

But splitting that functionality up on the server is the better long-term solution anyway...

Reply:
I'm actually going to dump this PR because it's easier than editing I reckon...

Reply:
No editing is as simple as pushing to the same branch

Reply:
But always make sure to create a branch first in your fork (don't push to main)